orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/PipelineTemplatePipelinePreprocessor.java

```diff
@@ -100,13 +102,19 @@ public PipelineTemplatePipelinePreprocessor(ObjectMapper pipelineTemplateObjectM
     Errors validationErrors = new Errors();
 
     TemplateConfiguration templateConfiguration = request.getConfig();
-    new V1TemplateConfigurationSchemaValidator().validate(templateConfiguration, validationErrors);
+
+    PipelineTemplate template = getPipelineTemplate(request, templateConfiguration);
+
+    new V1TemplateConfigurationSchemaValidator().validate(
+      templateConfiguration,
+      validationErrors,
+      new V1TemplateConfigurationSchemaValidator.SchemaValidatorContext(
+        template.getStages().stream().map(StageDefinition::getId).collect(Collectors.toList())
+      ));
```

Comment:
Pedantic, but lets drop the `);` to a new line to keep a consistent style.

Reply:
Done.